contracts/VaultV3.vy

```diff
@@ -253,10 +257,11 @@ def deposit(_amount: uint256, _recipient: address) -> uint256:
     amount: uint256 = _amount
 
     if amount == MAX_UINT256:
-        amount = ASSET.balanceOf(msg.sender)
+        amount = min(ASSET.balanceOf(msg.sender), self.depositLimit - self._totalAssets())
```

Comment:
I don't like depositing the remaining amount. I'd like to keep behavior always the same (if you pass MAX_UINT256 you are depositing your full balance. 

I think I would leave the if as it was then use the assert to be sure the deposit respects the deposit limit

Reply:
made the changes in [74708e9](https://github.com/jmonteer/yearn-vaults-v3/pull/38/commits/74708e94d2006ff048ad74bf149f8163fd732014)